From 6dc8c218d15e7e1745fb92de17f1794712fe496c Mon Sep 17 00:00:00 2001
From: zhenglixin <lixin.zhenglx@gmail.com>
Date: Sat, 27 Sep 2025 22:30:12 +0800
Subject: [PATCH 07/10] static link mbedtls

---
 Makefile.openwrt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.openwrt b/Makefile.openwrt
index 0106b15..74cab66 100644
--- a/Makefile.openwrt
+++ b/Makefile.openwrt
@@ -6,7 +6,7 @@ GNB_ES_LDFLAGS=-Wl,--gc-sections -L/usr/lib -pthread -flto
 
 ifeq ($(USE_MBEDTLS),1)
 CFLAGS += -DWITH_MBEDTLS=1
-GNB_ES_LDFLAGS += -lmbedtls -lmbedcrypto -lmbedx509
+GNB_ES_LDFLAGS += -Wl,-Bstatic -lmbedtls -lmbedcrypto -lmbedx509 -Wl,-Bdynamic
 else
 GNB_ES_LDFLAGS += -lssl -lcrypto
 endif
-- 
2.37.1 (Apple Git-137.1)

